source/f5_virtualserver.go

```diff
@@ -211,3 +219,10 @@ func (vs *f5VirtualServerSource) filterByAnnotations(virtualServers []*f5.Virtua
 
 	return filteredList, nil
 }
+
+func isVirtualServerReady(vs *f5.VirtualServer) bool {
+	normalizedStatus := strings.ToLower(vs.Status.Status)
```

Comment:
potential null pointer as `vs` is a reference 

The null check will help to avoid external-dns crashlooping.

Ideally the null check should be added in method 

```
func (vs *f5VirtualServerSource) endpointsFromVirtualServers
...

for _, virtualServer := range virtualServers {
   if virtualServer == nil {
            continue
        }
```

Reply:
Good catch but AFAICT the `vs` pointer here are explicitly [set earlier](https://github.com/kubernetes-sigs/external-dns/blob/dfe8e7806a91d3a1dcee17180234c7274e7b706b/source/f5_virtualserver.go#L113) before passed to the `endpointsFromVirtualServers` method, that would effectively mitigate any risk of encountering a nil pointer.

Reply:
Hi

Looks like missed that piece ;-) If server is empty and created like that, then you right.
```
virtualServer := &f5.VirtualServer{}
```

If the virtual server created like the one above, is there is a change is a chance to  throw a null pointer panic in helper function `isVirtualServerReady`, as vs.Status is not set?

```
vs.Status.Status 
vs.Status.VSAddress
```

Am I correct or missing something?

Reply:
Thanks for all the review comments, super helpful! 🙏🏻 

The `status` field of the `VirtualServer` object is defined as a non-pointer: https://github.com/F5Networks/k8s-bigip-ctlr/blob/508f686bf2886a4f95af0f3a822e6ed25bce2ec7/config/apis/cis/v1/types.go#L19. AFAICT this means that in this context, the fields of the `VirtualServerStatus` struct, will be set to their respective zero values if not explicitly initialized!